scipy/stats/_distn_infrastructure.py

```diff
@@ -3469,6 +3505,18 @@ def isf(self, q, *args, **kwds):
             return output[()]
         return output
 
+    def _unpack_loc_scale(self, theta):
+        try:
```

Comment:
This follows the logic of the implementation for `rv_continuous`, but I think it should unpack the other way around. If the tuple `theta` is missing values, this assumes that `loc` is always the last one, but really the _shapes_ are what should always be present. It doesn't matter here, since it's a private function and always called with all values present, so I won't change it in this PR. But if we decide to create `nllf` function to replace the misnamed `nnlf`, we should consider fixing this.